src/Modules/OrchardCore.Commerce/Services/ShoppingCartHelpers.cs

```diff
@@ -95,15 +95,16 @@ public class ShoppingCartHelpers : IShoppingCartHelpers
             totals = lines.CalculateTotals().ToList();
         }
 
-        for (var i = 0; i < totals.Count; i++)
+        // The values are rounded to avoid storing more precision than what the currency supports.
+        for (var i = 0; i < lines.Count; i++)
         {
-            var total = totals[i];
+            var line = lines[i];
 
-            // The value is rounded to avoid storing more precision than what the currency supports.
-            totals[i] = new Amount(Math.Round(total.Value, total.Currency.DecimalPlaces), total.Currency);
+            lines[i].LinePrice = line.LinePrice.GetRounded();
+            lines[i].UnitPrice = line.UnitPrice.GetRounded();
```

Comment:
I don't think you have to index here, you can access the same value via `line.LinePrice`. Also maybe this whole loop can be foreach instead of for?